.github/actions/npm-license-checker/src/checkLicenses.test.ts

```diff
@@ -153,4 +163,174 @@ describe('checkLicenses', () => {
       plainVertical: true
     })
   })
+
+  it('should handle detailsOutputPath correctly', async () => {
+    options.detailsOutputPath = './license-output.json'
+
+    licenseChecker.init.callsFake((opts: unknown, callback: InitCallback) => {
+      // @ts-expect-error - The first argument can be an error or null
+      callback(null, {})
+    })
+
+    await checkLicenses(licenseChecker, options, core)
+
+    assert.include(licenseChecker.init.firstCall.args[0], {
+      out: options.detailsOutputPath
+    })
+  })
+
+  it('should log license checker options', async () => {
+    licenseChecker.init.callsFake((opts: unknown, callback: InitCallback) => {
+      // @ts-expect-error - The first argument can be an error or null
+      callback(null, {})
+    })
+
+    await checkLicenses(licenseChecker, options, core)
+
+    assert.isTrue(core.info.calledOnce)
+    const infoMessage = core.info.firstCall.args[0]
+    assert.include(
+      infoMessage,
+      'Start checking licenses with the following options'
+    )
+    assert.include(infoMessage, options.startPath)
+    assert.include(infoMessage, 'MIT')
+  })
+
+  it('should handle customFields as an object', async () => {
+    const customFieldsObj: CustomFields = {
+      name: 'custom-name',
+      version: 'custom-version',
+      licenses: 'custom-license',
+      licenseText: 'custom-text'
+    }
+
+    options.customFields = customFieldsObj
+
+    licenseChecker.init.callsFake((opts: unknown, callback: InitCallback) => {
+      // @ts-expect-error - The first argument can be an error or null
+      callback(null, {})
+    })
+
+    await checkLicenses(licenseChecker, options, core)
+
+    assert.include(licenseChecker.init.firstCall.args[0], {
+      customFormat: customFieldsObj
+    })
+  })
+
+  it('should always set summary to true', async () => {
+    licenseChecker.init.callsFake((opts: unknown, callback: InitCallback) => {
+      // @ts-expect-error - The first argument can be an error or null
+      callback(null, {})
+    })
+
+    await checkLicenses(licenseChecker, options, core)
+
+    assert.include(licenseChecker.init.firstCall.args[0], {
+      summary: true
+    })
+  })
+
+  it('should handle undefined excludePackages', async () => {
+    delete options.excludePackages
```

Comment:
nit: Mutating objects like this is always a little risky in case there's something else that depends on the `options` variable being untouched. I typically find it better to do something like this:

```ts
const { excludePackages, ...optionsWithoutExcludePackages } = options;
```

And then do:

```
await checkLicenses(licenseChecker, optionsWithoutExcludePackages, core)
```